WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -139,6 +143,8 @@ def post_announcement
       body += " Check out the [#{comp.name} website](#{comp.website}) for more information and registration.";
     end
     create_post_and_redirect(title: title, body: body, author: current_user, world_readable: true)
+
+    comp.update!(announced_at: Time.now)
```

Comment:
Just a thought here: another way of doing this would be to give competitions an `announcement_post_id` which you could then use to get the announcement date by doing `comp.announcement.created_at`.

That said, I like this current solution as well.